icu4c/source/common/locid.cpp

```diff
@@ -326,12 +326,109 @@ struct Locale::Heap : public UMemory {
         }
     }
 
-    // Move should be done on the std::unique_ptr object that owns this.
+    // Move should be done on the owner of the pointer to this object.
     Heap(Heap&&) noexcept = delete;
 
     ~Heap() = default;
 };
 
+Locale::Payload::HeapPtr& Locale::Payload::HeapPtr::operator=(const Heap& other) {
+    U_ASSERT(type == eBOGUS);
+    UErrorCode status = U_ZERO_ERROR;
+    ptr = new Heap(other, status);
+    if (ptr == nullptr || U_FAILURE(status)) {
+        delete ptr;
+    } else {
+        type = eHEAP;
+    }
+    return *this;
+}
+
+Locale::Payload::HeapPtr& Locale::Payload::HeapPtr::operator=(HeapPtr&& other) noexcept {
+    U_ASSERT(type == eBOGUS);
+    ptr = other.ptr;
+    type = eHEAP;
+    other.type = eBOGUS;
+    return *this;
+}
+
+template <typename Fn0, typename Fn1, typename Fn2, typename... Args>
+auto Locale::Payload::visit(Fn0 fn0, Fn1 fn1, Fn2 fn2, Args... args) const {
```

Comment:
Please rename to clarify, e.g.,
- `Fn0 fn0` --> `BogusFn bogusFn`
- `Fn1 fn1` --> `NestFn nestFn`
- etc. or similar